main.go

```diff
@@ -72,6 +146,22 @@ func main() {
 		log.Fatalf("Unable to fetch SVID: %v", err)
 	}
 
+	// try read from cache
+	if cache_dir != "" {
+		if cache, err := NewCache(cache_dir); err != nil {
+			log.Warnf("Unable to create cache: %v", err)
+		} else {
+			if cached, ok := cache.Get(role_arn); ok {
```

Comment:
if err == null on line 151, cache will be nil and this line fails

Reply:
line 154 would not be executed if err != nil, right?

Reply:
yes, you're right.